src/rgw/driver/rados/rgw_sync.cc

```diff
@@ -1647,6 +1647,95 @@ class RGWMetaSyncShardCR : public RGWCoroutine {
         collect_children();
       } while (omapkeys->more && can_adjust_marker);
 
+      mdlog_marker = sync_marker.marker;
```

Comment:
what do the different "markers" (all set to the value of sync_marker.marker) intend to do?  a comment would be helpful here